doc/manual/manual.tex

```diff
@@ -6241,15 +6241,119 @@ \subsubsection{Crustal deformation}
 \end{figure}
 
 
+\subsubsection{Inner core convection}
+\label{sec:cookbooks-inner-core-convection}
 
+\textit{This section was contributed by Juliane Dannberg, and the model setup was insipired 
+by discussions with John Rudge.}
 
+This is an example for convection in the inner core of the Earth. It uses a spherical geometry 
+and is characterized by a single material with temperature dependent density, that makes it
+unstably stratified as temperature increases towards the center of the core. 
+The boundary conditions combine normal stress and normal velocity, and take into account the
+rate of phase change (melting/freezing) at the inner-outer core boundary. Gravity decreases 
+linearly from the boundary to zero at the center of the inner core. 
+The setup is analogous to the models described in \cite{Deguen2013}, and all material properties 
+are chosen in a way that the equations are non-dimensional.
 
+The required heating model and changes to the material model are implemented in a shared library
+(\texttt{inner\_core.cc}). In their non-dimensional form, the equations are
 
+\begin{align}
+  \label{eq:inner-core-1}
+  \nabla \cdot \sigma &=
+  -Ra T \mathbf g,
+  \\
+  \label{eq:inner-core-2}
+  \nabla \cdot \mathbf u &= 0,
+  \\
+  \label{eq:inner-core-3}
+  \left(\frac{\partial T}{\partial t} + \mathbf u\cdot\nabla T\right)
+  - \nabla^2 T
+  &=
+  H,
+\end{align}
 
+where $Ra$ is the Rayleigh number. 
+
+The mechanical boundary conditions for the inner core are
+tangential stress-free and continuity of the normal stress at the
+inner-outer core boundary. For the non-dimensional equations, that
+means that we can define a 'phase change number' $\mathcal{P}$ so that the
+normal stress at the boundary is $-\mathcal{P} u_r$ with the radial velocity
+$u_r$. This number characterizes the resistance to phase change at
+the boundary, with $\mathcal{P}\rightarrow\infty$ corresponding to infinitely slow
+melting/freezing (or a free slip boundary), and $\mathcal{P}\rightarrow0$ corresponding to
+instantaneous melting/freezing (or a zero normal stress, corresponding to an open boundary).
+
+In the weak form, this results in boundary conditions of the form
+of a surface integral:
+$$\int_S \mathcal{P} (\mathbf u \cdot \mathbf n) (\mathbf v \cdot \mathbf n) dS,$$
+with the normal vector $\mathbf n$.
+
+This phase change term is added to the matrix in the \texttt{inner\_core.cc} plugin by using a signal that 
+connects the function \verb!set_assemblers_phase_boundary!, which is only called once at the beginning of 
+the model run. It creates the new assembler \texttt{PhaseBoundaryAssembler} for the boundary faces of the Stokes 
```

Comment:
I think this is documented somewhere in the manual. Can you add a reference to that section?